src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -112,10 +115,18 @@ private static void setupWindows(Application application) {
 
         CommandManagerWindow commandManager = new CommandManagerWindow(CommandsFeature.getCommandManager());
         application.addWindowComponent("Command Manager", commandManager);
+        ExceedingCanvasCommandVisitor visitor = new ExceedingCanvasCommandVisitor(commandManager.getDrawPanel());
 
         CommandManagerWindowCommandChangeObserver windowObserver = new CommandManagerWindowCommandChangeObserver(
-                commandManager);
+                commandManager,visitor);
         CommandsFeature.getCommandManager().getChangePublisher().addSubscriber(windowObserver);
+        application.addComponentMenu(Canvas.class,"Canvas");
+        application.addComponentMenuElement(Canvas.class, "A4Canvas", (ActionEvent e) -> {
+            commandManager.setDrawPanel(new A4Canvas(300));});
+
+        application.addComponentMenuElement(Canvas.class, "BasicCanvas", (ActionEvent e) -> {
+            commandManager.setDrawPanel(new JPanelRectCanvas());});
```

Comment:
OK, but I do not like the idea of changing this panel (I prefer idea of a more abstract canvas).